### PR TITLE
feat(sast): add new python CDK policies

### PR DESCRIPTION
**Pull request by RabeaZr**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"
feat: added the first batch of policies
## Description
first batch of cdk policies
*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/cdk/checks/python/APIGatewayCacheEnable.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+metadata:
```
Comment (pazbechor) [inline checkov/cdk/checks/python/APIGatewayCacheEnable.yaml:1]: Check, performencewise - if it's better to do it as following: 
```
and: 
  - pattern: aws_cdk.aws_apigateway.Stage(<ANY>)
  - not_pattern: aws_cdk.aws_apigateway.Stage(<ANY>, cache_cluster_enabled=True, <ANY>)
```

Comment (pazbechor) [reply checkov/cdk/checks/python/APIGatewayCacheEnable.yaml:1]: @achiar99

Comment (pazbechor) [reply checkov/cdk/checks/python/APIGatewayCacheEnable.yaml:1]: @RabeaZr Notice - This comment is relevant to all the other patterns as well, so it's important.

File: checkov/cdk/checks/python/DAXEncryption.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  id: CKV_AWS_47
+  name: Ensure DAX is encrypted at rest (default is unencrypted)
+  category: ENCRYPTION
+scope:
+  languages:
+    - python
+definition:
+  pattern: aws_cdk.aws_dax.CfnCluster(<ANY>)
+  conditions:
+    - not_pattern: aws_cdk.aws_dax.CfnCluster(<ANY>, sse_specification=aws_cdk.aws_dax.CfnCluster.SSESpecificationProperty(<ANY>, enabled=True , <ANY>), <ANY>)
```
Comment (pazbechor) [inline checkov/cdk/checks/python/DAXEncryption.yaml:13]: Did you check this?: 
```
blabla = aws_cdk.aws_dax.CfnCluster.SSESpecificationProperty(enabled=True)
aws_cdk.aws_dax.CfnCluster(sse_specification=blabla)
```

File: checkov/cdk/checks/python/WAFEnabled.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  id: CKV_AWS_68
+  name: CloudFront Distribution should have WAF enabled
+  category: ENCRYPTION
+scope:
+  languages:
+    - python
+definition:
+  pattern: aws_cdk.aws_cloudfront.CfnDistribution(<ANY>)
+  conditions:
+    - not_pattern: 'aws_cdk.aws_cloudfront.CfnDistribution(<ANY>, distribution_config={<ANY>, "webAclId": <ANY> , <ANY>} , <ANY>)'
```
Comment (pazbechor) [inline checkov/cdk/checks/python/WAFEnabled.yaml:13]:
```suggestion
    - not_pattern: 'aws_cdk.aws_cloudfront.CfnDistribution(<ANY>, distribution_config={<ANY>, "webAclId": $ARG , <ANY>} , <ANY>)'
```

File: checkov/cdk/checks/python/ECRImmutableTags.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  id: CKV_AWS_51
+  name: Ensure ECR Image Tags are immutable
+  category: GENERAL_SECURITY
+scope:
+  languages:
+    - python
+definition:
+  pattern: aws_cdk.aws_ecr.Repository(<ANY>)
+  conditions:
+    - not_pattern: aws_cdk.aws_ecr.Repository(<ANY>, image_tag_mutability=aws_cdk.aws_ecr.TagMutability.IMMUTABLE, <ANY>)
```
Comment (pazbechor) [inline checkov/cdk/checks/python/ECRImmutableTags.yaml:13]: WDYT about this one? 
```suggestion
  pattern: aws_cdk.aws_ecr.Repository(<ANY>, image_tag_mutability=$ARG, <ANY>)
  conditions:
    - metavariable: $ARG
    - not_pattern: aws_cdk.aws_ecr.TagMutability.IMMUTABLE

```

File: checkov/cdk/checks/python/LambdaFunctionLevelConcurrentExecutionLimit.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+metadata:
```
Comment (pazbechor) [inline checkov/cdk/checks/python/LambdaFunctionLevelConcurrentExecutionLimit.yaml:1]: Notice that you also can implement the adding reserved_concurrent_executions as following, although I think it's less 'programatic' way, we should here what @tsmithv11 has to say regarding that
```
class LambdaStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        # Define the Lambda function using the low-level CfnFunction
        function = CfnFunction(self, "ExampleLambdaFunction",
                               runtime="python3.8",
                               handler="index.handler",
                               code=Code.from_asset("lambda").asset_path,  # Directory containing your Lambda code
                               function_name="ExampleLambdaFunction"
                               )

        # Set the reserved concurrency using CfnConcurrency
        reserved_concurrency = core.CfnResource(self, "ExampleLambdaFunctionConcurrency",
                                                type="AWS::Lambda::Concurrency",
                                                properties={
                                                    "FunctionName": function.function_name,
                                                    "ReservedConcurrentExecutions": 100
                                                })

        # Ensure concurrency settings are applied after the function is created
        reserved_concurrency.node.add_dependency(function)


app = core.App()
LambdaStack(app, "LambdaStack")
```

File: checkov/cdk/checks/python/LambdaInVPC.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    - python
+definition:
+  patterns:
+    or:
+      - pattern: aws_cdk.aws_lambda.Function(<ANY>)
+        conditions:
+          - not_pattern: aws_cdk.aws_lambda.Function(<ANY>, vpc=$VPC, <ANY>)
+      - pattern: aws_cdk.aws_sam.CfnFunction(<ANY>)
+        conditions:
+          - not_pattern: aws_cdk.aws_sam.CfnFunction(<ANY>, vpc_config=aws_cdk.aws_sam.CfnFunction.VpcConfigProperty(<ANY>), <ANY>)
```
Comment (pazbechor) [inline checkov/cdk/checks/python/LambdaInVPC.yaml:18]:
```suggestion
          - not_pattern: aws_cdk.aws_sam.CfnFunction(<ANY>, vpc_config=$ARG, <ANY>)
```

File: checkov/cdk/checks/python/LambdaDLQConfigured.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    - python
+definition:
+  patterns:
+    or:
+      - pattern: aws_cdk.aws_lambda.Function(<ANY>)
+        conditions:
+          - not_pattern: aws_cdk.aws_lambda.Function(<ANY>, dead_letter_queue=$ARG, <ANY>)
+      - pattern: aws_cdk.aws_lambda.CfnFunction(<ANY>)
+        conditions:
+          - not_pattern: aws_cdk.aws_lambda.CfnFunction(<ANY>, dead_letter_config={<ANY>}, <ANY>)
```
Comment (pazbechor) [inline checkov/cdk/checks/python/LambdaDLQConfigured.yaml:18]:
```suggestion
          - not_pattern: aws_cdk.aws_lambda.CfnFunction(<ANY>, dead_letter_config=$ARG, <ANY>)
```

File: checkov/cdk/checks/python/LambdaDLQConfigured.yaml [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+      - pattern: aws_cdk.aws_sam.CfnFunction(<ANY>)
+        conditions:
+          - not_pattern: aws_cdk.aws_sam.CfnFunction(<ANY>, dead_letter_queue=aws_cdk.aws_sam.CfnFunction.DeadLetterQueueProperty(<ANY>), <ANY>)
```
Comment (pazbechor) [inline checkov/cdk/checks/python/LambdaDLQConfigured.yaml:21]:
```suggestion
          - not_pattern: aws_cdk.aws_sam.CfnFunction(<ANY>, dead_letter_queue=$ARG, <ANY>)
```

File: checkov/cdk/checks/python/WAFEnabled.yaml (hunk 3 above)
Comment (pazbechor) [reply checkov/cdk/checks/python/WAFEnabled.yaml:13]: Also, Pls try it like this:
```suggestion
    - not_pattern: 'aws_cdk.aws_cloudfront.CfnDistribution(<ANY>, distribution_config={"webAclId": $ARG} , <ANY>)'
```

**Review by achiar99 — APPROVED** [review]

**Review by pazbechor — APPROVED** [review]

**Pull request merged**